jetty-core/jetty-http2/jetty-http2-common/src/main/java/org/eclipse/jetty/http2/internal/HTTP2Flusher.java

```diff
@@ -106,6 +108,8 @@ public boolean append(HTTP2Session.Entry entry)
         try (AutoLock ignored = lock.lock())
         {
             closed = terminated;
+            if (closed instanceof HpackException.SessionException)
+                closed = null;
             if (closed == null)
             {
                 entries.offer(entry);
```

Comment:
Can't the return true at line 117/121 below be done here (after the debug below) rather than testing again for `closed == null` outside the lock?

Reply:
Done.